src/ShareModal.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+import React from 'react';
+import ModalHeader from './ModalHeader';
+import ModalWithFooter from './ModalWithFooter';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import type {IntlShape} from 'react-intl';
+import {ReactComponent as ShareIcon} from './svg/Share.svg'
+
+import './ShareModal.scss';
+
+type ShareCompleteModalProps = {
+    intl: IntlShape,
+    show: boolean,
+    onCancel: () => void,
+    onConfirm: () => void
+};
+
+class ShareCompleteModal extends React.Component<ShareCompleteModalProps, {}> {
+    static defaultProps = {
+        show: false,
+        onCancel: () => {}
+    }
+
+    copyURL = () => {
+        // Copy the URL to the clipboard, see:
+        // https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText
+        if (navigator.clipboard) {
+            navigator.clipboard.writeText(document.location.href);
+        }
+    }
+
+    componentDidMount() {
+        this.copyURL();
+    }
+
+    /* istanbul ignore next */
+    handleKeyDown = (event: KeyboardEvent) => {
+        if (event.key === ' ' || event.key === 'Enter') {
+            event.preventDefault();
+            this.copyURL();
+        }
+    }
+
+    render () {
+        const buttonProperties = [{
+            label: this.props.intl.formatMessage({id: 'ShareModal.cancel'}),
+            onClick: this.props.onCancel,
+            isPrimary: false
+        }];
+
+        return(
+            <ModalWithFooter
+                show={this.props.show}
+                focusOnOpenSelector={'#ShareModal-copy'}
+                focusOnCloseSelector={'.ProgramBlockEditor__program-deleteAll-button'}
+                ariaLabel={this.props.intl.formatMessage({ id: 'ConfirmDeleteAllModal.title' })}
+                onClose={this.props.onCancel}
+                buttonProperties={buttonProperties}
+            >
+                <ModalHeader
+                    id='ShareModal'
+                    title={this.props.intl.formatMessage({
+                        id: 'ShareModal.title'
+                    })}
+                >
+                    <ShareIcon aria-hidden="true"/>
+                </ModalHeader>
+
+                <div className='ShareModal__content'>
+                    <p><FormattedMessage id='ShareModal.description1' /></p>
+                    <p><FormattedMessage id='ShareModal.description2' /></p>
+
+                    <div className='ShareModal__form'>
+                        <div className='ShareModal__form__URL__container'>
```

Comment:
Was there a reason why you used a div to show the URL instead of text field? I think it would make sense to be able to manually highlight URL in the container if users want to. Also, by using text field, we don't need to apply custom styles for the container. Here's a suggestion:

```
<input className='ShareModal__form__URL' type="text" readOnly={true} value={document.location.href} />
```

Reply:
If we decide to just use the text field I suggested, we can remove the ShareModal__form__URL__container and its child div, and just render the input element sibling of the share button.

Reply:
I was concerned that a text field would give the impression that the value was editable, or that people would mistakenly copy a partial URL manually.  We should discuss in the meeting tomorrow.

Reply:
We reviewed this with Sepi and went with a read only text input.